src/commands/addDataPlaneApis.ts

```diff
@@ -3,32 +3,43 @@
 import { IActionContext } from "@microsoft/vscode-azext-utils";
 import * as vscode from 'vscode';
 import { DataPlaneAccount } from "../azure/ApiCenter/ApiCenterDataPlaneAPIs";
+import { TelemetryClient } from "../common/telemetryClient";
+import { TelemetryEvent, TelemetryProperties } from "../common/telemetryEvent";
 import { ext } from "../extensionVariables";
 import { UiStrings } from "../uiStrings";
-export async function getDataPlaneApis(context: IActionContext): Promise<any | void> {
-    const endpointUrl = await vscode.window.showInputBox({ title: UiStrings.AddDataPlaneRuntimeUrl, ignoreFocusOut: true });
-    if (!endpointUrl) {
-        return;
-    }
-    const clientid = await vscode.window.showInputBox({ title: UiStrings.AddDataPlaneClientId, ignoreFocusOut: true });
-    if (!clientid) {
-        return;
+export namespace ConnectDataPlaneApi {
+    export async function getDataPlaneApis(context: IActionContext): Promise<any | void> {
+        const endpointUrl = await vscode.window.showInputBox({ title: UiStrings.AddDataPlaneRuntimeUrl, ignoreFocusOut: true });
+        if (!endpointUrl) {
+            return;
+        }
+        const clientid = await vscode.window.showInputBox({ title: UiStrings.AddDataPlaneClientId, ignoreFocusOut: true });
+        if (!clientid) {
+            return;
+        }
+        const tenantid = await vscode.window.showInputBox({ title: UiStrings.AddDataPlaneTenantId, ignoreFocusOut: true });
+        if (!tenantid) {
+            return;
+        }
+        ConnectDataPlaneApi.sendDataPlaneApiTelemetry(endpointUrl, clientid, tenantid);
+        ConnectDataPlaneApi.setAccountToExt(endpointUrl, clientid, tenantid);
+        ext.dataPlaneTreeItem.refresh(context);
     }
-    const tenantid = await vscode.window.showInputBox({ title: UiStrings.AddDataPlaneTenantId, ignoreFocusOut: true });
-    if (!tenantid) {
-        return;
+    export function sendDataPlaneApiTelemetry(runtimeUrl: string, clientId: string, tenantId: string) {
+        const properties: { [key: string]: string; } = {};
+        properties[TelemetryProperties.dataPlaneRuntimeUrl] = runtimeUrl;
+        properties[TelemetryProperties.dataPlaneTenantId] = tenantId;
+        properties[TelemetryProperties.dataPlaneClientId] = clientId;
+        TelemetryClient.sendEvent(TelemetryEvent.openUrlFromDataPlane, properties);
```

Comment:
This event is sent when a new data plane intance is added? `openUrl` is weird...

Reply:
currently the `openurl` only work for `addDataPlaneApis`, and `addDataPlaneApis` has three entry points that ui, command palette and deeplink.

Reply:
rename to `addDataPlaneFromUrl` for better understanding

Reply:
distinct with UI and DeepLink